kgforge/specializations/models/rdf/directory_service.py

```diff
@@ -61,30 +66,47 @@ def resolve_context(self, iri: str) -> Dict:
     def generate_context(self) -> Dict:
         return self._generate_context()
 
-    def _build_shapes_map(self) -> Dict:
-        query = """
-            PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
-            PREFIX sh: <http://www.w3.org/ns/shacl#>
-            SELECT ?type ?shape WHERE {
-                { ?shape sh:targetClass ?type .}
-                UNION {
-                    SELECT (?shape as ?type) ?shape WHERE {
-                        ?shape a sh:NodeShape .
-                        ?shape a rdfs:Class
-                    }
-                }
-            } ORDER BY ?type"""
+    def _build_shapes_map(self) -> Tuple[Dict, Dict, Dict]:
+        query = build_shacl_query(
+            defining_property_uri=self.NXV.shapes,
+            deprecated_property_uri=OWL.deprecated,
+            optional=True,
+            search_in_graph=True,
+            context=self.context,
+        )
         res = self._graph.query(query)
-        return {row["type"]: row["shape"] for row in res}
+        class_to_shape = {}
+        shape_to_defining_resource = {}
+        shape_to_named_graph = {}
+        defining_resource_to_named_graph = {}
+        for row in res:
+            class_to_shape[URIRef(row["type"])] = URIRef(row["shape"])
```

Comment:
small question: why to use an `URIRef` object instead of the `uri` as it is?

Reply:
also, the types can be loaded in shrink or expanded form, I think this should be uniformized here, using the `self.context.expand()` method

Reply:
>also, the types can be loaded in shrink or expanded form, I think this should be uniformized here, using the self.context.expand() method

The directory service gets a sparql result so all uris are already expanded. The directory service on the other hand needs that.

>small question: why to use an URIRef object instead of the uri as it is?

To catch as early as possible malformed URI

Reply:
I just confirmed, the directory service could also get the shrink versions of the uris, the expansion needs to happen there too

Reply:
Okay. Still surprised that a SPARQL query results returned curies. Can you print past here the SPARQL result json output you are getting ?

Reply:
```
(rdflib.term.URIRef('bmo:ExtractionTargetsConfiguration'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/extractiontargetsconfiguration/shapes/ExtractionTargetsConfigurationShape'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/extractiontargetsconfiguration'), rdflib.term.URIRef('shapes/datashapes/core/extractiontargetsconfiguration.json'))
(rdflib.term.URIRef('https://neuroshapes.org/BrainSlicing'), rdflib.term.URIRef('https://neuroshapes.org/dash/brainslicing/shapes/BrainSlicingShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/brainslicing'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/bbp/shapes_jsonld_expanded/datashapes/core/brainslicing.ttl'))
(rdflib.term.URIRef('https://neuroshapes.org/IonChannelGene'), rdflib.term.URIRef('https://neuroshapes.org/dash/ionchannelgene/shapes/IonChannelGeneShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/ionchannelgene'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/shapes_jsonld_expanded/datashapes/core/ionchannelgene.ttl'))
(rdflib.term.URIRef('https://neuroshapes.org/DetailedCircuit'), rdflib.term.URIRef('https://neuroshapes.org/dash/detailedcircuit/shapes/DetailedCircuitShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/detailedcircuit'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/shapes_jsonld_expanded/datashapes/core/detailedcircuit.ttl'))
(rdflib.term.URIRef('bmo:CellTypeAssignmentAnnotation'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/celltypeassignmentannotation/shapes/CellTypeAssignmentAnnotationShape'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/celltypeassignmentannotation'), rdflib.term.URIRef('shapes/datashapes/core/celltypeassignmentannotation.json'))
(rdflib.term.URIRef('http://www.w3.org/ns/prov#Collection'), rdflib.term.URIRef('https://neuroshapes.org/dash/collection/shapes/CollectionShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/collection'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/bbp/shapes_jsonld_expanded/datashapes/core/collection.ttl'))
(rdflib.term.URIRef('https://neuroshapes.org/Transformation'), rdflib.term.URIRef('https://neuroshapes.org/dash/transformation/shapes/TransformationShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/transformation'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/shapes_jsonld_expanded/datashapes/core/transformation.ttl'))
(rdflib.term.URIRef('https://neuroshapes.org/SubjectCollection'), rdflib.term.URIRef('https://neuroshapes.org/dash/subjectcollection/shapes/SubjectCollectionShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/subjectcollection'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/bbp/shapes_jsonld_expanded/datashapes/core/subjectcollection.ttl'))
(rdflib.term.URIRef('https://neuroshapes.org/TemplateImageData'), rdflib.term.URIRef('https://neuroshapes.org/dash/templateimagedata/shapes/TemplateImageDataShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/templateimagedata'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/bbp/shapes_jsonld_expanded/datashapes/core/templateimagedata.ttl'))
(rdflib.term.URIRef('https://neuroshapes.org/ElectrophysiologyFeature'), rdflib.term.URIRef('https://neuroshapes.org/dash/electrophysiologyfeature/shapes/ElectrophysiologyFeatureShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/electrophysiologyfeature'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/bbp/shapes_jsonld_expanded/datashapes/core/electrophysiologyfeature.ttl'))
(rdflib.term.URIRef('https://neuroshapes.org/TraceFeature'), rdflib.term.URIRef('https://neuroshapes.org/dash/tracefeature/shapes/TraceFeatureShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/tracefeature'), rdflib.term.URIRef('shapes/datashapes/core/tracefeature.json'))
(rdflib.term.URIRef('bmo:PSPcapacitanceVoltage'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/pspcapacitancevoltage/shapes/PSPcapacitanceVoltageShape'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/pspcapacitancevoltage'), rdflib.term.URIRef('shapes/datashapes/core/pspcapacitancevoltage.json'))
(rdflib.term.URIRef('http://schema.org/QuantitativeValue'), rdflib.term.URIRef('https://neuroshapes.org/dash/quantitativevalue/shapes/QuantitativeValueShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/quantitativevalue'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/shapes_jsonld_expanded/datashapes/core/quantitativevalue.ttl'))
(rdflib.term.URIRef('bmo:RawSpikeTrain'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/rawspiketrain/shapes/RawSpikeTrainShape'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/rawspiketrain'), rdflib.term.URIRef('shapes/datashapes/core/rawspiketrain.json'))
(rdflib.term.URIRef('bmo:SynapticTypeMapping'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/synaptictypemapping/shapes/SynapticTypeMappingShape'), rdflib.term.URIRef('https://bbp.epfl.ch/shapes/dash/synaptictypemapping'), rdflib.term.URIRef('shapes/datashapes/core/synaptictypemapping.json'))
(rdflib.term.URIRef('https://neuroshapes.org/NdRaster'), rdflib.term.URIRef('https://neuroshapes.org/dash/ndraster/shapes/NdRasterShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/ndraster'), rdflib.term.URIRef('shapes/datashapes/core/ndraster.json'))
(rdflib.term.URIRef('https://neuroshapes.org/ReconstructedWholeBrainCell'), rdflib.term.URIRef('https://neuroshapes.org/dash/reconstructedwholebraincell/shapes/ReconstructedWholeBrainCellShape'), rdflib.term.URIRef('https://neuroshapes.org/dash/reconstructedwholebraincell'), rdflib.term.URIRef('shapes/test_tmp_dir/ontologies/shapes_jsonld_expanded/datashapes/core/reconstructedwholebraincell.ttl'))
```

Reply:
you can see that many of them are in CURIE form

Reply:
Okay.  I made the change. I think this a symptom that the context was not completely taken into account when the directory service is building the graph